src/Compilers/CSharp/Portable/Symbols/TypeParameterBounds.cs

```diff
@@ -75,14 +79,42 @@ private TypeParameterBounds()
         /// When computing the deduced type we don't perform this abstraction. We keep the original constraint T.
         /// Deduced base type is used to check that consistency rules are satisfied.
         /// </summary>
-        public readonly TypeSymbol DeducedBaseType;
+        public readonly TypeSymbol? DeducedBaseType;
     }
 
     internal static class TypeParameterBoundsExtensions
     {
-        internal static bool IsSet(this TypeParameterBounds boundsOpt)
+        internal static bool HasValue(this TypeParameterBounds? boundsOpt, bool canIgnoreNullableContext)
+        {
+            if (boundsOpt == TypeParameterBounds.Unset)
+            {
+                return false;
+            }
+            if (boundsOpt == null)
+            {
+                return true;
+            }
+            return canIgnoreNullableContext || !boundsOpt.IgnoresNullableContext;
+        }
+
+        // Returns true if bounds was updated with value.
+        // Returns false if bounds already had a value with expected 'IgnoresNullableContext'
```

Comment:
nit: consider wording this as "a value with *sufficient* IgnoresNullableContext". Similar on the next line.

Something to indicate that when the caller requests a bounds that ignores nullable context, the symbol may provide a bounds that doesn't ignore nullable context.

Reply:
Thanks. I'll update the comment in a subsequent PR.